compose/snippets/src/main/java/com/example/compose/snippets/animations/AnimationSnippets.kt

```diff
@@ -709,6 +723,104 @@ private fun AnimationSpecKeyframe() {
     // [END android_compose_animations_spec_keyframe]
 }
 
+@OptIn(ExperimentalAnimationSpecApi::class)
+@Composable
+private fun AnimationSpecKeyframeWithSpline() {
+    // [START android_compose_animation_spec_keyframes_with_spline]
+    val offset by animateOffsetAsState(
+        targetValue = Offset(300f, 300f),
+        animationSpec = keyframesWithSpline {
+            durationMillis = 6000
+            Offset(0f, 0f) at 0
+            Offset(150f, 200f) atFraction 0.5f
+            Offset(0f, 100f) atFraction 0.7f
+        }
+    )
+    // [END android_compose_animation_spec_keyframes_with_spline]
+}
+
+@Suppress("PrimitiveInCollection")
+@OptIn(ExperimentalAnimationSpecApi::class)
+@Preview
+@Composable
+private fun OffsetKeyframeWithSplineDemo() {
+    val points = remember { mutableStateListOf<Offset>() }
+    val offsetAnim = remember {
+        androidx.compose.animation.core.Animatable(
+            Offset.Zero,
+            Offset.VectorConverter
+        )
+    }
+    val density = LocalDensity.current
+
+    BoxWithConstraints(
+        Modifier.fillMaxSize().drawBehind {
+            drawPoints(
+                points = points,
+                pointMode = PointMode.Lines,
+                color = Color.LightGray,
+                strokeWidth = 4f,
+                pathEffect = PathEffect.dashPathEffect(floatArrayOf(30f, 20f))
+            )
+        }
+    ) {
+        val minDimension = minOf(maxWidth, maxHeight)
+        val size = minDimension / 4
+
+        val sizePx = with(density) { size.toPx() }
+        val widthPx = with(density) { maxWidth.toPx() }
+        val heightPx = with(density) { maxHeight.toPx() }
+
+        val maxXOff = (widthPx - sizePx) / 2f
+        val maxYOff = heightPx - (sizePx / 2f)
+
+        Box(
+            Modifier.align(Alignment.TopCenter)
+                .offset { offsetAnim.value.round() }
+                .size(size)
+                .background(Color.Red, RoundedCornerShape(50))
+                .onPlaced { points.add(it.boundsInParent().center) }
+        )
+
+        LaunchedEffect(Unit) {
+            delay(1000)
+            while (isActive) {
+                offsetAnim.animateTo(
+                    targetValue = Offset.Zero,
+                    animationSpec =
+                    keyframesWithSpline {
+                        durationMillis = 4400
+
+                        // Increasingly approach the halfway point moving from side to side
+                        repeat(4) {
```

Comment:
```suggestion
                        for (i in 1 .. 4) {
```

Reply:
To me, using `for` with a range is slightly more readable than using `repeat` and initializing `i` separately.